### PR TITLE
Minor optimisations

### PR DESCRIPTION
**Pull request by tsmgeek**
- Optimisations to using switch statements in CommandController.
- Assign getView() to local variable ReportConfigController

### CODE REVIEW

File: web/app/view/dialog/ReportConfigController.js [hunk 1]
```diff
@@ -27,22 +27,23 @@ Ext.define('Traccar.view.dialog.ReportConfigController', {
 
     onSaveClick: function (button) {
         var eventType;
-        this.getView().callingPanel.deviceId = this.lookupReference('deviceField').getValue();
-        this.getView().callingPanel.groupId = this.lookupReference('groupField').getValue();
+        var callingPanel = this.getView().callingPanel;
```
Comment (tananaev) [inline web/app/view/dialog/ReportConfigController.js]: Please define all variable in the first line. If assignment is too long, split definition and assignment.

File: web/app/view/dialog/CommandController.js [hunk 2]
```diff
@@ -42,67 +42,62 @@ Ext.define('Traccar.view.dialog.CommandController', {
...
-        }
-
-        if (record.get('type') === 'custom') {
-            value = this.lookupReference('paramCustom').getValue();
-            record.set('attributes', {
-                data: value
-            });
+        
+
+        switch(record.get('type')){
```
Comment (tananaev) [inline web/app/view/dialog/CommandController.js]: Have you run `jshint` and `jscs` on your code? If not, please do and fix any issues.

File: web/app/view/dialog/ReportConfigController.js [hunk 3]
```diff
@@ -26,23 +26,25 @@ Ext.define('Traccar.view.dialog.ReportConfigController', {
     ],
 
     onSaveClick: function (button) {
-        var eventType;
-        this.getView().callingPanel.deviceId = this.lookupReference('deviceField').getValue();
-        this.getView().callingPanel.groupId = this.lookupReference('groupField').getValue();
+        var eventType,callingPanel;
```
Comment (tananaev) [inline web/app/view/dialog/ReportConfigController.js]: Are you sure you have check formatting after fixing issue? Please double-check everything before submitting pull request.

**Comment by tsmgeek** [comment]
Moved from NB to VSC, could not get jsint/jscs to work with NB.
Are you going to migrate to eslint, i saw a note saying jscs is merging into it?

**Comment by tananaev** [comment]
Possible, but I haven't looked into it. Feel free to send a pull request.

File: web/app/view/dialog/ReportConfigController.js (hunk 3 above)
Comment (tsmgeek) [inline web/app/view/dialog/ReportConfigController.js]: I have both packages you mention installed and I can see them working as now if i force an error it shows notes from both.

Comment (tananaev) [inline web/app/view/dialog/ReportConfigController.js]: There is a space missing. Not sure why `jscs` didn't pick it up.

**Comment by tsmgeek** [comment]
Can I add the following to .gitignore to catch anyone pushing these created by the minify script?

>web/app.min.js
>sencha-error-*

**Comment by tsmgeek** [comment]
Ive put in the correct jscs command `requireSpaceAfterComma` to flag up space after comma.

**Comment by tananaev** [comment]
I think we should just add `web/app.min.js` and `sencha-error-*` to `.gitignore` file.

Does `eslint` replace both `jshint` and `jscs`? If yes, then we should probably remove other configs.

Another idea. Add check to the travis-ci, so we can get checks as a part of pull requests.

**Comment by tsmgeek** [comment]
It seems that eslint replaces jscs and considering it does so much it can also replace jshint.
Ive used `polyjuice` to convert jshint config to eslint. not removed other configs yet in case you want to try it first.

https://www.sitepoint.com/comparison-javascript-linting-tools/

**Comment by tananaev** [comment]
Official documentation says that `.eslintrc` is deprecated:

http://eslint.org/docs/user-guide/migrating-from-jscs

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**